publisher-sdk/src/main/java/com/criteo/publisher/dependency/LazyDependency.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.dependency
+
+/**
+ * Lazily provide and cache the dependency that can be build from the given supplier.
+ *
+ * If you've got a dependency on a type that is particularly expensive to produce, you can use lazy dependency to defer
```

Comment:
I would avoid using the second person in code documentation (as it's not consistent with the tone of voice in the rest of the project).

Reply:
Done